File: DSA/Manav-Rajput/Day45_question2.java
class Solution {
    public String orderlyQueue(String s, int k) {
        if (k == 1) {
            String smallest = s;
            String rotated = s;
            for (int i = 0; i < s.length(); i++) {
                rotated = rotated.substring(1) + rotated.charAt(0);
                if (rotated.compareTo(smallest) < 0) {
                    smallest = rotated;
                }
            }
            return smallest;
        } else {
            char[] chars = s.toCharArray();
            Arrays.sort(chars);
            return new String(chars);
        }
    }
}

File: DSA/Arshdeep-Kaur/Day 45/Day45_Question2.cpp
class Solution {
public:
    string orderlyQueue(string s, int k) {
        if (k > 1) {
            sort(s.begin(), s.end());
            return s;
        }

        string s_min = s;

        for(int i=0; i<s.length(); ++i) {
            s = s.substr(1) + s[0];
            s_min = min(s_min, s);
        }

        return s_min;
    }
};
